### ISSUE TITLE
PP a vtl file with no content throws an NPE

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#18051]
If  you have a vtl file with no content in it then push publishing that vtl file fails with the error:

```
 ERROR pusher.PushPublisher: null
java.lang.NullPointerException
at com.dotcms.publisher.pusher.PushPublisher.process(PushPublisher.java:200) [dotcms_4.3.3_999999.jar:?]
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:127) [dotcms_4.3.3_999999.jar:?]
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:27) [dotcms_4.3.3_999999.jar:?]
at com.dotcms.publisher.business.PublisherQueueJob.execute(PublisherQueueJob.java:186) [dotcms_4.3.3_999999.jar:?]
at org.quartz.core.JobRunShell.run(JobRunShell.java:223) [dot.quartz-all-1.8.6_2.jar:?]
at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549) [dot.quartz-all-1.8.6_2.jar:?]
[30/01/20 07:22:00:047 GMT] ERROR publishing.PublisherAPIImpl: null
com.dotcms.publishing.DotPublishingException
at com.dotcms.publisher.pusher.PushPublisher.process(PushPublisher.java:330) ~[dotcms_4.3.3_999999.jar:?]
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:127) [dotcms_4.3.3_999999.jar:?]
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:27) [dotcms_4.3.3_999999.jar:?]
at com.dotcms.publisher.business.PublisherQueueJob.execute(PublisherQueueJob.java:186) [dotcms_4.3.3_999999.jar:?]
at org.quartz.core.JobRunShell.run(JobRunShell.java:223) [dot.quartz-all-1.8.6_2.jar:?]
at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549) [dot.quartz-all-1.8.6_2.jar:?]
[30/01/20 07:22:00:047 GMT] ERROR business.PublisherQueueJob: Unable to publish Bundle: null
com.dotcms.publishing.DotPublishingException
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:139) ~[dotcms_4.3.3_999999.jar:?]
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:27) ~[dotcms_4.3.3_999999.jar:?]
at com.dotcms.publisher.business.PublisherQueueJob.execute(PublisherQueueJob.java:186) [dotcms_4.3.3_999999.jar:?]
at org.quartz.core.JobRunShell.run(JobRunShell.java:223) [dot.quartz-all-1.8.6_2.jar:?]
at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:549) [dot.quartz-all-1.8.6_2.jar:?]
Caused by: com.dotcms.publishing.DotPublishingException
at com.dotcms.publisher.pusher.PushPublisher.process(PushPublisher.java:330) ~[dotcms_4.3.3_999999.jar:?]
at com.dotcms.publishing.PublisherAPIImpl.publish(PublisherAPIImpl.java:127) ~[dotcms_4.3.3_999999.jar:?]
... 4 more
[30/01/20 07:22:46:213 GMT] ERROR websocket.SystemEventsWebSocketEndPoint: java.util.concurrent.TimeoutException
java.io.IOException: java.util.concurrent.TimeoutException
at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.startMessageBlock(WsRemoteEndpointImplBase.java:282) ~[tomcat-websocket.jar:8.0.18]
at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.sendPing(WsRemoteEndpointImplBase.java:174) ~[tomcat-websocket.jar:8.0.18]
at org.apache.tomcat.websocket.WsRemoteEndpointBase.sendPing(WsRemoteEndpointBase.java:55) ~[tomcat-websocket.jar:8.0.18]
at com.dotcms.rest.api.v1.system.websocket.SystemEventsWebSocketEndPoint.doPing(SystemEventsWebSocketEndPoint.java:116) ~[dotcms_4.3.3_999999.jar:?]
at com.dotcms.rest.api.v1.system.websocket.SystemEventsWebSocketEndPoint.processPingPongQueue(SystemEventsWebSocketEndPoint.java:95) ~[dotcms_4.3.3_999999.jar:?]
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_171]
at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) [?:1.8.0_171]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_171]
at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) [?:1.8.0_171]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_171]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_171]
at java.lang.Thread.run(Thread.java:748) [?:1.8.0_171]
Caused by: java.util.concurrent.TimeoutException
at org.apache.tomcat.websocket.FutureToSendHandler.get(FutureToSendHandler.java:99) ~[tomcat-websocket.jar:8.0.18]
at org.apache.tomcat.websocket.WsRemoteEndpointImplBase.startMessageBlock(WsRemoteEndpointImplBase.java:275) ~[tomcat-websocket.jar:8.0.18]
```

To test, upload a vtl file into dotcms, and then in the `/assets` file system, wipe out the content of that file, then try to PP it.

from: 
https://dotcms.zendesk.com/agent/tickets/100502

**Comment by wezell** [comment]
I think we should ship with `CONTENT_ALLOW_ZERO_LENGTH_FILES=true` and see if this resolves the PP issue.  My gut is that we should allow 0 length files out of the box.

**Comment by erickgonzalez** [comment]
PR: https://github.com/dotCMS/core/pull/18665

**Comment by bryanboza** [comment]
Fixed, tested on master // Postgres // FF

**Comment by jcastro-dotcms** [comment]
The official code changes have been included in the dotCMS 5.2.8.4 LTS release. Code changes were part of the dotCMS 5.3.8.4 LTS release already.
